File: find_xuanzang.py
content = open(r"D:\pyproject\hero_workshop_react\src\data\npcs.ts", encoding="utf-8").read()
idx = content.find("贫僧启程在即")
if idx >= 0:
    print("Found at:", idx)
    print(repr(content[idx:idx+400]))

File: find_xuanzang.mjs
import fs from 'fs';

const f = fs.readFileSync("D:/pyproject/hero_workshop_react/src/data/npcs.ts", "utf8");
const i = f.indexOf("id: 'changan_xuanzang'");
const end = f.indexOf("  },", f.indexOf("personalItem:", i) + 30);
// Look for next "  }," after personalItem
console.log("Context around personalItem xuanzang:", JSON.stringify(f.substring(end-60, end+10)));
